### DISCUSSION TITLE
Cmdlet and Parameter not found - ConvertTo-PnPPage, TransformationOnPrem

### DISCUSSION THREAD

**Discussion by RuthChurchman** [pnp/modernization#600] [Q&A]
I'm attempting to migrate and modernize publishing pages from SharePoint 2010 On Prem to SP Online Communication site using this sample script described here: []([https://docs.microsoft.com/en-us/sharepoint/dev/transform/modernize-userinterface-site-pages-powershell#read-publishing-page-in-on-premises-sharepoint-and-create-the-modern-page-in-sharepoint-online](https://docs.microsoft.com/en-us/sharepoint/dev/transform/modernize-userinterface-site-pages-powershell#read-publishing-page-in-on-premises-sharepoint-and-create-the-modern-page-in-sharepoint-online))

As described in article, I have installed the legacy PnP version to be compatible with SP2010. 

In the script 2nd line TransformationOnPrem has error "Parameter cannot be found", and in 3rd line ConvertTo-PnPPage is "not recognized": 

"PS C:\Users\****> $source = Connect-PnPOnline http://****/IT/ruth/Migration -TransformationOnPrem -CurrentCredentials -ReturnConnection
Connect-PnPOnline : A parameter cannot be found that matches parameter name 'TransformationOnPrem'.
At line:1 char:61
 ... POnline http://****/IT/ruth/Migration -TransformationOnPrem -Curren ...

 CategoryInfo          : InvalidArgument: (:) [Connect-PnPOnline], ParameterBindingException
 FullyQualifiedErrorId : NamedParameterNotFound,PnP.PowerShell.Commands.Base.ConnectOnline

PS C:\Users\****> ConvertTo-PnPPage -Identity "category.aspx" -PublishingPage -TargetConnection $target -LogVerbose -LogType File -LogFolder c:\temp -Connection $source
ConvertTo-PnPPage : The term 'ConvertTo-PnPPage' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
again.
At line:1 char:1
ConvertTo-PnPPage -Identity "category.aspx" -PublishingPage -TargetCo ...
CategoryInfo          : ObjectNotFound: (ConvertTo-PnPPage:String) [], CommandNotFoundException
FullyQualifiedErrorId : CommandNotFoundException"
"

I am using SharePointPnPPowerShellOnline 3.29.2101.0 on Powershell 5.1.19041.906
I have admin rights on SharePoint2010 and SPO.

What am I missing? Do I need to install something extra?

**Comment by jansenbe** [comment]
@RuthChurchman : please create an issue in https://github.com/pnp/pnpframework/issues as Page Transformation is part of PnP Framework.

**Reply by RuthChurchman** [reply]
@jansenbe Thanks for letting me know. I have created the issue there. Please feel free to delete this. Thanks
